src/terminal/index.ts

```diff
@@ -284,18 +288,10 @@ class TerminalWidget extends Widget {
    */
   protected onResize(msg: ResizeMessage): void {
     if (!this.isAttached || !this.isVisible) {
```

Comment:
you only need to check `isVisible` here, a detached widget will be `!isVisible`.